src/models/iceflow/SIA2D/SIA2D_utils.jl

```diff
@@ -141,9 +144,13 @@ function SIA2D(H::Matrix{R}, simulation::SIM, t::R; batch_id::Union{Nothing, I}
     Fy = .-avg_x(D) .* dSdy_edges 
 
     #  Flux divergence
-    @tullio dH[i,j] := -(diff_x(Fx)[pad(i-1,1,1),pad(j-1,1,1)] / Δx + diff_y(Fy)[pad(i-1,1,1),pad(j-1,1,1)] / Δy) 
```

Comment:
@ChrisRackauckas I see you're working in the out-of-place version of the code. Since Enzyme supports in-place, why not directly switch to the `SIA!` function? That one doesn't have Tullio, and in the end it's the one we want to use, to avoid memory allocations.

Reply:
How do I switch to that? That would make it easier.

Reply:
You just have to change this line to `Huginn.SIA2D!`:

https://github.com/ODINN-SciML/ODINN.jl/blob/9aa8c9b2fb357a509f4bdc4e03aff5b98210b922/src/simulations/functional_inversions/functional_inversion_utils.jl#L147